### PR TITLE
Optimize Service Migration test to avoid mutiple pod restarts

### PR DESCRIPTION
**Pull request by anpanigr**
Merged all three tests methods to one to avoid multiple scale down/up to reduce test execution time
Add Persistent Store Migration check

External Jenkin ( in progress)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/6553/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 1]
```diff
@@ -350,12 +349,12 @@ private void runJmsClientOnAdminPod(String action, String queue) {
 
   /*
    * Verify the JMS Server Runtime through REST API.
-   * Get the JMSServer Runtime ClusterJmsServer@managed-server2 found on
-   * specified managed server.
+   * Get the sepecific JMSServer Runtime on specified managed server.
```
Comment (bhavaniravichandran) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:352]: correct specific spelling

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 2]
```diff
@@ -206,125 +217,113 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
...
-    assertDoesNotThrow(() -> copyFileToPod(domainNamespace,
-        adminServerPodName, "",
-        Paths.get(RESOURCE_DIR, "jms", "JmsSendReceiveClient.java"),
-        Paths.get(destLocation)));
-    runJavacInsidePod(adminServerPodName, domainNamespace, destLocation);
+    assertTrue(checkJtaRecoveryServiceRuntime("managed-server2", 
+         "managed-server2", "true"), 
+         "JTARecoveryService@managed-server2 is not on managed-server2 before migration");
 
+    // Send persistent messages to both FileStore abd JDBCStore based
```
Comment (bhavaniravichandran) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:246]: FileStore abd JDBCStore
                ^^and

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 3]
```diff
@@ -206,125 +217,113 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
...
 
   /**
-   * Verify JMS Service is migrated to an available active server.
-   * Here the JMS messages are stored in Filestore on PV
+   * Verify JMS/JTA Service is migrated to an available active server.
    */
   @Test
-  @DisplayName("Verify JMS Service migration with FileStore")
-  void testMiiJmsServiceMigrationWithFileStore() {
+  @DisplayName("Verify JMS/JTA Service migration with File(JDBC) Store")
```
Comment (bhavaniravichandran) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:223]: File and JDBC Store

**Review by bhavaniravichandran — APPROVED** [review]
minor comments

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 4]
```diff
@@ -206,125 +217,113 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
...
         adminServerPodName, "",
         Paths.get(RESOURCE_DIR, "jms", "JmsSendReceiveClient.java"),
         Paths.get(destLocation)));
     runJavacInsidePod(adminServerPodName, domainNamespace, destLocation);
 
-    assertTrue(checkJmsServerRuntime("managed-server2"),
-         "JMSService@managed-server2 is on managed-server2 before migration");
+    assertTrue(checkJmsServerRuntime("ClusterJmsServer@managed-server2", 
+         "managed-server2"),
+        "ClusterJmsServer@managed-server2 is on managed-server2 before migration");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:236]: If its false, shouldn't the message be ClusterJmsServer@managed-server2 is not on managed-server2 before migration.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java [hunk 5]
```diff
@@ -206,125 +217,113 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
...
-
-    // Restart the managed server(2) if shutdown by previous test method
-    // Make sure that JMS server runtime JMSService@managed-server2 is
-    // hosted on managed server 'managed-server2'
-    restartManagedServer("managed-server2");
-    assertTrue(checkJmsServerRuntime("managed-server2"),
-         "JMSService@managed-server2 is on managed-server2 before migration");
+    assertTrue(checkJmsServerRuntime("JdbcJmsServer@managed-server2", 
+        "managed-server2"),
+        "JdbcJmsServer@managed-server2 is on managed-server2 before migration");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java:240]: same as above

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
Also run just the It class couple more times on jenkins.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 1 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 4 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java (hunk 5 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiServiceMigration.java]: done

**Review by vanajamukkara — APPROVED** [review]
LG

**Pull request merged**